core/domain/action_registry.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for actions."""
+
+import os
+import pkgutil
+
+import feconf
+
+
+class Registry(object):
+    """Registry of all actions."""
+
+    # Dict mapping action IDs to instances of the actions.
+    _actions = {}
+
+    @classmethod
+    def get_all_action_ids(cls):
+        """Get a list of all action IDs."""
+        return feconf.ALLOWED_ACTION_IDS
+
+    @classmethod
+    def set_actions(cls, actions):
+        cls._actions = actions
+
+    @classmethod
+    def _refresh(cls):
+        cls._actions.clear()
+
+        all_action_ids = cls.get_all_action_ids()
+
+        # Assemble all paths to the actions.
+        extension_paths = [
+            os.path.join(feconf.ACTIONS_DIR, action_id)
+            for action_id in all_action_ids]
+
+        # Crawl the directories and add new action instances to the
+        # registry.
+        for loader, name, _ in pkgutil.iter_modules(path=extension_paths):
+            module = loader.find_module(name).load_module(name)
+            clazz = getattr(module, name)
+
+            ancestor_names = [
+                base_class.__name__ for base_class in clazz.__bases__]
+            if 'BaseLearnerActionSpec' in ancestor_names:
+                cls._actions[clazz.__name__] = clazz()
+
+    @classmethod
+    def get_all_actions(cls):
+        """Get a list of instances of all actions."""
```

Comment:
Incomplete docstrings (here and everywhere else) - please check that your docstrings for the methods are consistent and have the following format: args, returns, raises. But, I did notice that many of the methods' docstrings in the codebase do not follow this pattern (mostly the old ones).

Reply:
Done.